GitUI/CommandsDialogs/SettingsDialog/Pages/FormBrowseRepoSettingsPage.cs

```diff
@@ -1,78 +1,502 @@
+                listView.LargeImageList = imageList;
+
+                foreach (var image in ResourceImagesProvider.Images)
+                {
+                    imageList.Images.Add(image.Key, image.Value);
+
+                    listView.Items.Add(image.Key, image.Key);
+                }
+
+                listView.Click += (object sender, EventArgs e) =>
+                {
+                    editorService.CloseDropDown();
+                };
+
+                editorService.DropDownControl(listView);
+
+                if (listView.SelectedItems.Count > 0)
+                {
+                    return listView.SelectedItems[0].Text;
+                }
+
+                return value;
+            }
         }
 
-        private void cboTerminal_Enter(object sender, EventArgs e)
+        [TypeConverter(typeof(PropertySorter))]
+        private sealed class ShellViewModel : IShell, INotifyPropertyChanged, ICheckable
         {
-            _cboTerminalPreviousIndex = cboTerminal.SelectedIndex;
+            private const string ShellCategory = "Shell";
+
+            public static ShellViewModel Create()
+            {
+                return new ShellViewModel();
+            }
+
+            public static ShellViewModel Create(IShell shell)
+            {
+                return new ShellViewModel
+                {
+                    Id = shell.Id,
+                    Name = shell.Name,
+                    Icon = shell.Icon,
+                    Command = shell.Command,
+                    Arguments = shell.Arguments,
+                    Default = shell.Default,
+                    Enabled = shell.Enabled
+                };
+            }
+
+            private string _name;
+            private string _icon;
+            private string _command;
+            private string _arguments;
+            private bool _default;
+            private bool _enabled;
+
+            public event PropertyChangedEventHandler? PropertyChanged;
+
+            private ShellViewModel()
+            {
+                Id = Guid.NewGuid();
+                _name = ShellConstants.DefaultName;
+                _icon = ShellConstants.DefaultIcon;
+                _command = ShellConstants.DefaultCommand;
+                _arguments = ShellConstants.DefaultArguments;
+                _default = false;
+                _enabled = true;
+            }
+
+            [Browsable(false)]
+            public Guid Id { get; private set; }
+
+            [Category(ShellCategory)]
+            [PropertyOrder(0)]
+            public string Name
+            {
+                get
+                {
+                    return _name;
+                }
+
+                set
+                {
+                    if (_name == value)
+                    {
+                        return;
+                    }
+
+                    _name = value;
+
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Name)));
```

Comment:
These can be made significantly simpler with the use of `CallerMemberName` attribute, have a look at https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged?view=netframework-4.6.1#examples